components/authz-service/storage/postgres/migration/legacy/iam_migration_test.go

```diff
@@ -0,0 +1,384 @@
+package legacy
+
+import (
+	"context"
+	"database/sql"
+	"net/url"
+	"testing"
+
+	"github.com/gofrs/uuid"
+	_ "github.com/golang-migrate/migrate/database/postgres" // make driver available
+	_ "github.com/golang-migrate/migrate/source/file"       // make source available
+
+	constants_v2 "github.com/chef/automate/components/authz-service/storage/postgres/migration/legacy/constants/v2"
+	"github.com/chef/automate/lib/logger"
+	"github.com/golang-migrate/migrate"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	v2DefaultAndLegacyPolicyCount = 13 // owner, editor, viewer, ingest + legacy
+	v2DefaultRoleCount            = 4  // owner, editor, viewer, ingest
+	v2DefaultProjectCount         = 2  // ~~All Projects~~, (unassigned)
+	pgURL                         = "postgres://postgres@127.0.0.1:5432/authz_test?sslmode=disable"
+)
+
+const resetDatabaseStatement = `DROP SCHEMA public CASCADE;
+CREATE SCHEMA public;
+GRANT ALL ON SCHEMA public TO postgres;
+GRANT ALL ON SCHEMA public TO public;`
+
+func TestMigrateToV2(t *testing.T) {
```

Comment:
These tests are ported from https://github.com/chef/automate/blob/master/components/authz-service/server/v2/policy_test.go#L2121

The logic to port legacy policies was not altered, so some tests that have high overhead weren't moved (combining 3 legacy policies for cfgmt, for ex).

We _have_ altered `migration_status` logic and those tests need to be taken care of here or in another ticket.